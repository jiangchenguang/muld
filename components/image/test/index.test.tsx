import * as React from 'react';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import { render, cleanup, fireEvent } from '@testing-library/react';
import Image from '..';

describe('Image', () => {
    afterEach(cleanup);
    it('click event', () => {
        const onClick = jest.fn();
        const { container } = render(<Image onClick={onClick} />);
        const panel = container.firstChild;
        expect(panel).toHaveClass('mul-image');
        fireEvent.click(panel!);
        expect(onClick).toHaveBeenCalled();
    });

    it('load event', () => {
        const onLoad = jest.fn();
        const { container } = render(
            <Image src="https://img.yzcdn.cn/vant/cat.jpeg" onLoad={onLoad} />,
        );
        const img = container.querySelector('img');
        fireEvent.load(img!);
        expect(onLoad).toHaveBeenCalled();
    });

    it('error event', () => {
        const onError = jest.fn();
        const { container } = render(
            <Image src="https://img.yzcdn.cn/vant/cat.jpeg" onError={onError} />,
        );
        const img = container.querySelector('img');
        fireEvent.error(img!);
        expect(onError).toHaveBeenCalled();
    });

    it('show-loading prop', () => {
        const { asFragment } = render(<Image showLoading={false} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('show-error prop', () => {
        const { asFragment, container } = render(
            <Image src="https://img.yzcdn.cn/vant/cat.jpeg" showError={false} />,
        );
        const img = container.querySelector('img');
        fireEvent.error(img!);
        expect(asFragment()).toMatchSnapshot();
    });

    it('loading-icon prop', () => {
        const { asFragment } = render(<Image loadingIcon="success" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('radius prop', () => {
        const { asFragment } = render(
            <Image radius="3" src="https://img.yzcdn.cn/vant/cat.jpeg" />,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
