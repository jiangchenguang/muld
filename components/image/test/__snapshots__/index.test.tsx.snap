// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image loading-icon prop 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm IvyVS mul-image"
  >
    <img
      class="mul-image__img"
      style="object-fit: fill;"
    />
    <div
      class="mul-image__loading"
    >
      <i
        class="sc-fznyAO fSDXHL mul-image__loading-icon mul-image__loading-icon-success"
        content="F0C8"
        name="success"
        style="font-size: 20px;"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Image radius prop 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm IvyVS mul-image"
    style="overflow: hidden; border-radius: 3px;"
  >
    <img
      class="mul-image__img"
      src="https://img.yzcdn.cn/vant/cat.jpeg"
      style="object-fit: fill;"
    />
    <div
      class="mul-image__loading"
    >
      <i
        class="sc-fznyAO bGzndu mul-image__loading-icon mul-image__loading-icon-photo"
        content="F09D"
        name="photo"
        style="font-size: 20px;"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Image show-error prop 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm IvyVS mul-image"
  />
</DocumentFragment>
`;

exports[`Image show-loading prop 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm IvyVS mul-image"
  >
    <img
      class="mul-image__img"
      style="object-fit: fill;"
    />
  </div>
</DocumentFragment>
`;
